### PR TITLE
[receiver/awss3receiver] Introduce the AWS S3 Receiver

### PR DESCRIPTION
**Pull request by adcharre**
**Description:** Initial skeleton implementation of the AWS S3 receiver described in issue #30750.
Full implementation will follow in future PRs.

**Link to tracking Issue:** #30750

**Testing:** -

**Documentation:** Initial README added.

### CODE REVIEW

File: receiver/awss3receiver/README.md
Comment (crobert-1) [inline receiver/awss3receiver/README.md:17]: Can you fill out the configuration details here, and also include an example config?

**Review by crobert-1 — COMMENTED** [review]
Since this change includes tests, can you add goleak tests as well? It's a recently added requirement for writing new components 👍 

[There's an entry now in the docs here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components).

**Review by atoulme — APPROVED** [review]

File: receiver/awss3receiver/README.md
Comment (crobert-1) [inline receiver/awss3receiver/README.md:20]: Can you include which options are required and which are optional?

File: receiver/awss3receiver/metadata.yaml
Comment (crobert-1) [inline receiver/awss3receiver/metadata.yaml:10]:
```suggestion
```

File: receiver/awss3receiver/metadata.yaml
Comment (crobert-1) [inline receiver/awss3receiver/metadata.yaml:7]: Should this be considered a part of the `contrib` distribution, even if it's not included in releases?

Comment (adcharre) [reply receiver/awss3receiver/metadata.yaml:7]: @crobert-1 is this a question for me? I suspect this was just copy'n'pasted from another receiver, happy to be told what to put here.

Comment (mx-psi) [reply receiver/awss3receiver/metadata.yaml:7]: I would leave the `distributions` field empty for now

**Comment by adcharre** [comment]
@crobert-1 / @mx-psi  - Anything left for me to do here before this PR can be merged?

**Comment by crobert-1** [comment]
> @crobert-1 / @mx-psi - Anything left for me to do here before this PR can be merged?

Nothing that I'm aware of. Since the `ready to merge` label is added, it's just waiting for a maintainer to merge. 

(I didn't get a chance to look through thoroughly enough to approve, but I don't have any notable concerns here.)

**Pull request merged**
